cnd/src/network.rs

```diff
@@ -88,13 +91,33 @@ impl Swarm {
     pub async fn initiate_communication(
         &self,
         id: LocalSwapId,
-        peer: DialInformation,
+        DialInformation {
+            peer_id,
+            address_hint,
+        }: DialInformation,
         role: Role,
         digest: SwapDigest,
         identities: Identities,
     ) -> anyhow::Result<()> {
         let mut guard = self.inner.lock().await;
-        guard.initiate_communication(id, peer, role, digest, identities)
+
+        // best effort attempt to establish a connection to the other party
+        if let Some(address_hint) = address_hint {
```

Comment:
Correct me if I'm wrong but if the user of cnd POSTS a swap as Bob but includes an IP address (easy mistake to make) the we will dial from Bob's node to Alice's node as well as listen and receive a connection request from Alice's node so we have two TCP connections. This doesn't seem like what we want?

Reply:
I think it is not that critical.

libp2p can easy handle multiple connections to the same peer and if we move the logic to the outside here, it is truly a concern of cnd instead of the comit lib.

I can guard it with role == Alice if you want.